### DISCUSSION TITLE
Generate syntax diagram

### DISCUSSION THREAD

**Discussion by vanillajonathan** [pest-parser/pest#803] [Ideas]
This could be done with a diagram generator such as:
- https://tabatkins.github.io/railroad-diagrams/
- https://www.bottlecaps.de/rr/ui

**Comment by nu11ptr** [comment]
Or the rust `railroad` crate. See here for my new crate: https://crates.io/crates/pest_railroad_gen. 

Would there be interest in a PR that integrates this directly into the Pest project/repo? I really don't want to maintain it and I think it would make a nice "out of the box" feature (as a separate lib/binary crate). Thoughts?

**Reply by tomtau** [reply]
Awesome work! I'd need to look in detail whether it'd make sense for it to go to the main repo, but it looks like something that could definitely live under the pest-parser GH org.
I imagine it could be nice to integrate in the existing pest tooling, such as VSCode extension to have a command to generate it on a separate tab or if it could compile to WASM, perhaps there could be an option for it in the web fiddle editor.

**Reply by nu11ptr** [reply]
Sounds good. I'm open to whatever makes sense, but yeah the VSCode extension would be great as a "preview" option or something on a grammar.
